acceptance/builder_test.go

```diff
@@ -0,0 +1,367 @@
+
+	rand.Seed(time.Now().UTC().UnixNano())
+
+	// FIXME: this is for development speed we need to comment out before production !
+	//h.MakeAndCopyLifecycle(t, "linux", builderBinaryDir)
+	h.DockerBuild(t,
+		builderImage,
+		builderDockerContext,
+		h.WithArgs("--build-arg", fmt.Sprintf("cnb_platform_api=%s", api.Platform.Latest())),
+	)
+	// FIXME: this is for development speed we need to comment out before production !
+	//defer h.DockerImageRemove(t, builderImage)
+
+	spec.Run(t, "acceptance-builder", testBuilder, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testBuilder(t *testing.T, when spec.G, it spec.S) {
+	// .../cmd/lifecycle/builder.go#Args
+	when("called with arguments", func() {
+		it("errors", func() {
+			command := exec.Command(
+				"docker",
+				"run",
+				"--rm",
+				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+				builderImage,
+				"some-arg",
+			)
+			output, err := command.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "failed to parse arguments: received unexpected arguments"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	// .../cmd/lifecycle/builder.go#Privileges
+	when("running as a root", func() {
+		it("errors", func() {
+			command := exec.Command(
+				"docker",
+				"run",
+				"--rm",
+				"--user",
+				"root",
+				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+				builderImage,
+			)
+			output, err := command.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "failed to build: refusing to run as root"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("error on reading Data", func() {
+		// .../cmd/lifecycle/builder.go#readData
+		when("read buildpack group file", func() {
+			when("no default group toml file in default location", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to read buildpack group"
+					h.AssertStringContains(t, string(output), expected)
+				})
+
+			})
+
+			//TODO: check some output file for this case not except any error message
+			when("empty group toml file", func() {
+				it("success", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/empty_group.toml",
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+						builderImage,
+					)
+					_, err := command.CombinedOutput()
+					//print(string(output), err)
+					h.AssertNil(t, err)
+					//expected := "failed to read buildpack order file"
+					//h.AssertStringContains(t, string(output), expected)
```

Comment:
I think you could check for a metadata.toml file to show that the build was completed.

Reply:
For 2 positive test I am checking metadata.toml now.